gnmvidispine/vidispine_api.py

```diff
@@ -428,7 +428,10 @@ def request(self,path,method="GET",matrix=None,query=None,body=None, accept='app
     @staticmethod
     def _escape_for_query(value):
         if isinstance(value,basestring):
-            toprocess = value.encode("UTF-8")
+            try:
+                toprocess = value.encode("UTF-8")
+            except UnicodeDecodeError:
```

Comment:
try replacing this with value.encode("UTF-8","backslashescape").decode("UTF-8"), or the other way around, that should make the try/except redundant. if you pass through something that causes utf-8 errors (as here) it may well blow up somewhere else

Reply:
toprocess = value.decode("UTF-8","backslashescape").encode("UTF-8") works for this but then the test called test_param_list_unicode breaks thus: - 

Error
Traceback (most recent call last):
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/tests/test_vidispine_api.py", line 392, in test_param_list_unicode
    response = VSApi._get_param_list("keyname",u"arséne wenger est allée en vacances. Häppy hølidåys")
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/gnmvidispine/vidispine_api.py", line 443, in _get_param_list
    return map(lambda item: "{0}={1}".format(key, VSApi._escape_for_query(item)), toprocess)
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/gnmvidispine/vidispine_api.py", line 443, in <lambda>
    return map(lambda item: "{0}={1}".format(key, VSApi._escape_for_query(item)), toprocess)
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/gnmvidispine/vidispine_api.py", line 431, in _escape_for_query
    toprocess = value.decode("UTF-8","backslashescape").encode("UTF-8")
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/gnmvidispine/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 3: ordinal not in range(128)

Do you still want it done with your line?

Reply:
@Dala0 try the other way around (encode first, with backslashescape, then decode)

Reply:
It is possible you actually meant me to try toprocess = value.decode("UTF-8").encode("UTF-8","backslashescape") That breaks the same test thus: -

Error
Traceback (most recent call last):
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/tests/test_vidispine_api.py", line 392, in test_param_list_unicode
    response = VSApi._get_param_list("keyname",u"arséne wenger est allée en vacances. Häppy hølidåys")
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/gnmvidispine/vidispine_api.py", line 443, in _get_param_list
    return map(lambda item: "{0}={1}".format(key, VSApi._escape_for_query(item)), toprocess)
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/gnmvidispine/vidispine_api.py", line 443, in <lambda>
    return map(lambda item: "{0}={1}".format(key, VSApi._escape_for_query(item)), toprocess)
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/gnmvidispine/vidispine_api.py", line 431, in _escape_for_query
    toprocess = value.decode("UTF-8").encode("UTF-8","backslashescape")
  File "/Volumes/LaCie/Repositories/portal3vm/work/gnmvidispine/gnmvidispine/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 3: ordinal not in range(128)

Reply:
@Dala0 does it still break if you encode first then decode? these methods are terribly named!

Reply:
Yes. That breaks both tests.

@fredex42

Reply:
😞 what happens if you put "backslashescape" into the decode instead of the encode?

Reply:
That is shown breaking in the first error I posted above.

@fredex42

Reply:
I assume you mean this?

toprocess = value.decode("UTF-8","backslashescape")

That breaks both tests.

@fredex42

Reply:
gah. ok, screw it. i'll accept the original version